tasks/src/lib.rs

```diff
@@ -80,22 +135,114 @@ pub trait Task: 'static {
 }
 
 /// Scheduler managing enqueued tasks and routing incoming responses to active tasks.
+#[derive(Debug)]
 pub struct Scheduler {
     flow: ClientFlow,
     waiting_tasks: TaskMap,
     active_tasks: TaskMap,
     tag_generator: TagGenerator,
+    capabilities: Vec1<Capability<'static>>,
+    server_id: Option<Vec<(IString<'static>, NString<'static>)>>,
```

Comment:
Current server id parameters (from the `ID` extension). If found it useful to expose it to users?